### PR TITLE
Assign cell fix

### PR DESCRIPTION
**Pull request by seichhorn**
The old assigncellfov had several bugs, I think I've fixed them in this version

### CODE REVIEW

File: merlin/analysis/segment.py [hunk 1]
```diff
@@ -168,32 +168,40 @@ def _construct_fov_tree(self, tiledPositions: pandas.DataFrame,
...
+                        element.get_bounding_box(), obj=element)
+
+    def _return_overlapping_cells(self, currentCell, cells: List):
+        areas = [currentCell.intersection(x) for x in cells]
+        overlapping = [cells[i] for i, x in enumerate(areas) if x > 0]
+        benchmark = currentCell.intersection(currentCell)
+        contained = [x for x in overlapping if
+                     x.intersection(currentCell) == benchmark]
+        if len(contained) > 1:
+            overlapping = []
```
Comment (emanuega) [inline merlin/analysis/segment.py]: Can you move the functionality of this function and the remove_overlapping_cells function into a utility class and write unit tests to test the edge cases so we can be sure this works as expected this time? This could possibly fit into the util/spatialfeature.py module.

**Comment by seichhorn** [comment]
I've changed as you requested, please take a look at the testing I perform and make sure it covers the cases you'd like.

File: test/test_spatialfeature.py [hunk 2]
```diff
@@ -193,3 +207,73 @@ def test_feature_contains_positions():
...
+            (p1 not in t4) and (p2 not in t4) and (p5 not in t4))
+    assert ((p5 in t5) and (p1 not in t5)
+            and (p2 not in t5) and (p3 not in t5) and (p4 not in t5))
+
+
+def test_remove_overlapping_cells():
+
+    def construct_graph(cells):
+        G = nx.Graph()
+        spatialIndex = rtree.index.Index()
```
Comment (emanuega) [inline test/test_spatialfeature.py]: This is nearly there. I would prefer that the construct_graph function be moved to spatialfeature.py too so that the code doesn't have to be repeated in the tests. This should be possible if the cells are preloaded before calling this function and if _construct_graph in segment.py is modified to take a list of cells and the fov boxes. This will then allow you to expand the tests to make sure that the test cells are allocated to the correct fov.

Comment (seichhorn) [reply test/test_spatialfeature.py]: I've addressed this now

File: merlin/analysis/segment.py [hunk 3]
```diff
@@ -168,32 +170,20 @@ def _construct_fov_tree(self, tiledPositions: pandas.DataFrame,
...
-    def _run_analysis(self, fragmentIndex):
-
-        featureDB = self.get_feature_database()
+                        element.get_bounding_box(), obj=element)
 
+    def _construct_graph(self):
+        G = nx.Graph()
         spatialIndex = rtree.index.Index()
         allFOVs = self.dataSet.get_fovs()
+        allFOVs = list(range(2))
```
[the remark this answers is not in the document]
Comment (seichhorn) [reply merlin/analysis/segment.py]: Ha, yeah, that was when I was debugging something, fixed.
